sorc/ncep_post.fd/MDL2P.f

```diff
@@ -532,6 +533,80 @@ SUBROUTINE MDL2P(iostatusD3D)
+                 ! when none actually exists on the adjacent model levels.
+                 ! 
+                 ! In the code below, we search for the condition in
+                 ! which this artifact occurs. Then, the previously 
+                 ! interpolated value of Qrain is simply replaced by the 
+                 ! value from the overlying (subfreezing) model level, 
+                 ! where Qrain is typically zero. The same approach is 
+                 ! extended to snow and graupel, thus preserving the total
+                 ! precipitation mixing ratio locally.
+                 ! 
+                 ! Note that the code below does not eliminate all supercooled
+                 ! rain on pressure levels, nor should it. Situations in which
+                 ! supercooled rain is permitted include:
+                 !  i) The situation described by (1)-(3) above, but with
+                 !     rain water also present on the overlying (subfreezing) model 
+                 !     level. This may correspond to freezing drizzle that falls
+                 !     through a melting level (becoming ordinary drizzle). In the
+                 !     code below, the value of Qrain on the target pressure level 
+                 !     will simply be replaced by the value from the overlying 
+                 !     model level. 
+                 ! ii) Classic freezing rain situations, characterized by a
+                 !     deep (model-resolved) surface-based refreezing layer. In
+                 !     the code below, the temperature check contained in the third 
+                 !     "IF" statement will evaluate to false.
+
+                 IF (MODELNAME == 'FV3R' .OR. MODELNAME == 'GFS') THEN
+                   IF ( (QR1(I,J) > 0.) .AND. (TSL(I,J) <= 273.15) ) THEN ! Supercooled rain water is present on this pressure level
+                     IF ( (T(I,J,LL-1) <= 273.15) .AND. &  ! Temp on overlying model level is subfreezing, and
+                          (T(I,J,LL) > 273.15) ) THEN      ! temp on underlying model level is above freezing (i.e., near melting level)
+                          ! Replace the interpolated mixing ratios of rain, snow, and graupel 
+                          ! on this pressure level with the values from the overlying (subfreezing) model level:
+                          print*,'Supercooled rain found at (I,J,P)=',I,J,SPL(LP)/100.
+                          print*,'...near melting level: above (T,P)=',T(I,J,LL-1),PMID(I,J,LL-1)/100.
+                          print*,'                       below (T,P)=',T(I,J,LL),PMID(I,J,LL)/100.
+                          print*,'.......using Qrain from above=',QQR(I,J,LL-1)
+                          print*,' '
+                          QR1(I,J) = QQR(I,J,LL-1)       
+                          QS1(I,J) = QQS(I,J,LL-1)                        
+                          QG1(I,J) = QQG(I,J,LL-1)                        
```

Comment:
QR1, QS1, and QG1 are values on the pressure level; QQR, QQS, QQG are the model-level values